### PR TITLE
handle nullable lists when normalizing

### PR DESCRIPTION
**Pull request by PatrykWalach**

### CODE REVIEW

File: libs/isograph-react/src/core/cache.ts [hunk 1]
```diff
@@ -127,8 +127,8 @@ type NetworkResponseValue =
   | NetworkResponseScalarValue
   | null
   | NetworkResponseObject
-  | NetworkResponseObject[]
-  | NetworkResponseScalarValue[];
+  | (NetworkResponseObject | null)[]
+  | (NetworkResponseScalarValue | null)[];
```
Comment (PatrykWalach) [inline libs/isograph-react/src/core/cache.ts:131]: I've noticed we expect lists to be not nullable

File: libs/isograph-react/src/core/cache.ts [hunk 2]
```diff
@@ -596,35 +601,29 @@ function normalizeNetworkResponseObject(
...
-): data is NetworkResponseScalarValue | NetworkResponseScalarValue[] {
-  // N.B. empty arrays count as empty arrays of linked fields.
+function isNullOrEmptyArray(data: unknown): data is never[] | null[] | null {
   if (Array.isArray(data)) {
     if (data.length === 0) {
-      return false;
+      return true;
     }
-    // This is maybe fixed in a new version of Typescript??
-    return (data as any).every((x: any) => isScalarOrEmptyArray(x));
```
Comment (PatrykWalach) [inline libs/isograph-react/src/core/cache.ts:621]: Keeping this as is would return true for list of scalars

File: libs/isograph-react/src/core/cache.ts [hunk 3]
```diff
@@ -461,17 +461,24 @@ function normalizeLinkedField(
     return existingValue !== null;
   }
 
-  if (isScalarButNotEmptyArray(networkResponseData)) {
+  if (
+    isScalarOrEmptyArray(networkResponseData) &&
+    !isNullOrEmptyArray(networkResponseData)
```
Comment (PatrykWalach) [inline libs/isograph-react/src/core/cache.ts:466]: 3. So I just do two checks here.

**Review by rbalicki2 — APPROVED** [review]
LGTM

**Pull request merged**